City-of-Helsinki/hel-fi-drupal-grants
# AU-930: Various robot tests

# [AU-930](https://helsinkisolutionoffice.atlassian.net/browse/AU-930)

# [AU-819](https://helsinkisolutionoffice.atlassian.net/browse/AU-819)



This PR is combined from two tickets, AU-930 and AU-819. Mostly introduces new tests.

- Modified the tests and pipeline to use similar variables as in `lomaketyokalu` (like environment, browser type)
- Improved public website tests
- Separated two different tests for application forms
- Fixed Azure devops pipeline setup
- Profile form template needed to have a class added for better selector targeting in tests
- Created robot tests for all profile types

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-0000_insert_correct_branch`
  * `make fresh`
* Run `make drush-cr`

## How to test


`make test-robot`, everything should pass
*Except* if you don't have "Yleisavustushakemus" published, then that test (number 5) will fail.

[AU-930]: https://helsinkisolutionoffice.atlassian.net/browse/AU-930?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


[AU-819]: https://helsinkisolutionoffice.atlassian.net/browse/AU-819?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
